From baa09c7e4c29d0aa0e0b20c20a5a947d28beb120 Mon Sep 17 00:00:00 2001
From: Gyuho Lee <leegyuho@amazon.com>
Date: Thu, 17 Oct 2019 13:02:49 -0700
Subject: --EKS-PATCH-- staging/*/apiserver: disable info level gRPC
 logging

gRPC balancer wrapper has (non-leveled) info logging on "Notify" call,
which means we are logging every single endpoint resolution. When the
cluster scales up, the logging for apiserver can dramatically
increase.

Since etcd v3 client already provides all the relevant information,
let's disable gRPC info logging to reduce the log size.

Signed-off-by: Jackson West <jgw@amazon.com>
---
 staging/src/k8s.io/apiserver/pkg/storage/etcd3/BUILD     | 1 +
 staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go | 6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/staging/src/k8s.io/apiserver/pkg/storage/etcd3/BUILD b/staging/src/k8s.io/apiserver/pkg/storage/etcd3/BUILD
index a233ea072ce..6beb983b634 100644
--- a/staging/src/k8s.io/apiserver/pkg/storage/etcd3/BUILD
+++ b/staging/src/k8s.io/apiserver/pkg/storage/etcd3/BUILD
@@ -82,6 +82,7 @@ go_library(
         "//vendor/go.etcd.io/etcd/clientv3:go_default_library",
         "//vendor/go.etcd.io/etcd/etcdserver/api/v3rpc/rpctypes:go_default_library",
         "//vendor/go.etcd.io/etcd/mvcc/mvccpb:go_default_library",
+        "//vendor/google.golang.org/grpc/grpclog:go_default_library",
         "//vendor/k8s.io/klog:go_default_library",
         "//vendor/k8s.io/utils/trace:go_default_library",
     ],
diff --git a/staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go b/staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go
index f2cf0e1b26c..f514c47d656 100644
--- a/staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go
+++ b/staging/src/k8s.io/apiserver/pkg/storage/etcd3/logger.go
@@ -18,13 +18,19 @@ package etcd3
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
 
 	"go.etcd.io/etcd/clientv3"
+	"google.golang.org/grpc/grpclog"
 	"k8s.io/klog"
 )
 
 func init() {
 	clientv3.SetLogger(klogWrapper{})
+
+	// overwrite gRPC logger, to discard all gRPC info-level logging
+	grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, os.Stderr, os.Stderr))
 }
 
 type klogWrapper struct{}
-- 
2.32.0

